bluesky/simulators.py

```diff
@@ -114,3 +121,401 @@ def check_limits(plan):
+    std_dev tuple for each message in the plan.
+
+    PARAMETERS
+    ----------
+    plan : generator.
+        The bluesky plan that the est_time is to be estimated for.
+
+    RETURN PARAMETERS
+    -----------------
+    out_time : list.
+        A list containing a msg/est_time/std_dev named tuple for each message in the plan.
+    '''
+    
+    out_time = []
+    for msg in plan:
+        est_time = obj_est(msg)
+        out_time.append(_MsgStats( msg, est_time.est_time, est_time.std_dev))
+
+    return out_time
+
+         
+
+def combine_est(est_time_1, est_time_2, method = 'sum'):
+    """
+    Returns the combination est_time/std_dev pairs et_1 and et_2.
+    This function returns the combination of est_time_1 and est_time_2, combined using the 
+    method definedby 'method'.
+
+    Parameters
+    ----------
+    est_time_1, est_time_2 : namedtuples.
+        The tuples containing the est_time/std_dev tuples to be combined.
+    method : string, optional.
+        The method to use for the combination of est_time_1 and est_time_2, default is 
+        'sum'.
+
+    Return Parameters
+    -----------------
+    out_time : namedtuple.
+        The combined est_time/std_dev tuple.
+    """
+    
+    if method == 'sum':
+        #adds the est_time, uses quadrature for std_dev.
+        out_time = _TimeStats(est_time_1.est_time + est_time_2.est_time, 
+                        (est_time_1.std_dev**2 + est_time_2.std_dev**2)**.5)
+
+    if method == 'subtract':
+        #adds the est_time, uses quadrature for std_dev.
+        out_time = _TimeStats(est_time_1.est_time - est_time_2.est_time, 
+                        (est_time_1.std_dev**2 + est_time_2.std_dev**2)**.5)
+
+    elif method == 'max':
+        #finds the max est_time, finds the potentially largest difference from the max_time
+        #as the std_dev.
+        max_time = max(est_time1.est_time, est_time2.est_time) 
+        pos_stdev = max(est_time1.est_time + est_time1.std_dev,
+                        est_time2.est_time + est_time2.std_dev )
+        neg_stdev = min(est_time1.est_time - est_time1.std_dev,
+                        est_time2.est_time - est_time2.std_dev )
+        max_stdev = max(abs(pos_stdev-max_time), abs(neg_stdev-max_time))
+        out_time = _TimeStats(max_time, max_stdev)
+
+    return out_time
+
+
+
+def obj_est(msg ):
+    """
+    Returns the est_time/std_dev pair for the object referenced in msg.
+    This function returns the est_time/std_dev pair for the object referenced in msg.obj and 
+    for the command type reference in msg.command.
+
+    Parameters
+    ----------
+    msg : message.
+        The first message of the group that was detected in plan_ETA.
+    
+    Return Parameters
+    -----------------
+    out_time : list.
+        The combined est_time/std_dev pair.
+
+    Updated Parameters
+    ------------------
+    _plan_history : dict.
```

Comment:
This should be promoted to a parameter and passed in by all callers.

Reply:
As mentioned above, this was done to avoid having the functions take this in as an argument and return a possibly modified version.